test/compound.jl

```diff
@@ -0,0 +1,79 @@
+using Random, Test, HDF5
+
+import HDF5.datatype
+import Base.unsafe_convert
+
+struct foo
+  a::Float64
+  b::String
+  c::String
+  d::Array{ComplexF64,2}
+  e::Array{Int64,1}
+end
+
+struct foo_hdf5
+  a::Float64
+  b::Cstring
+  c::NTuple{10, Cchar}
+  d::NTuple{9, ComplexF64}
+  e::HDF5.Hvl_t
+end
+
+function unsafe_convert(::Type{foo_hdf5}, x::foo)
+  foo_hdf5(x.a,
+           Base.unsafe_convert(Cstring, x.b),
+           ntuple(i -> x.c[i], length(x.c)),
+           ntuple(i -> x.d[i], length(x.d)),
+           HDF5.Hvl_t(convert(Csize_t, length(x.e)), convert(Ptr{Cvoid}, pointer(x.e)))
+          )
+end
+
+function datatype(::Type{foo_hdf5})
+  dtype = HDF5.h5t_create(HDF5.H5T_COMPOUND, sizeof(foo_hdf5))
+  HDF5.h5t_insert(dtype, "a", fieldoffset(foo_hdf5, 1), datatype(Float64))
+
+  vlenstr_dtype = HDF5.h5t_copy(HDF5.H5T_C_S1)
+  HDF5.h5t_set_size(vlenstr_dtype, HDF5.H5T_VARIABLE)
+  HDF5.h5t_set_cset(vlenstr_dtype, HDF5.H5T_CSET_UTF8)
+  HDF5.h5t_insert(dtype, "b", fieldoffset(foo_hdf5, 2), vlenstr_dtype)
+
+  fixedstr_dtype = HDF5.h5t_copy(HDF5.H5T_C_S1)
+  HDF5.h5t_set_size(fixedstr_dtype, 10 * sizeof(Cchar))
+  HDF5.h5t_set_cset(fixedstr_dtype, HDF5.H5T_CSET_UTF8)
+  HDF5.h5t_insert(dtype, "c", fieldoffset(foo_hdf5, 3), fixedstr_dtype)
+
+  hsz = HDF5.Hsize[3,3]
```

Comment:
Why [3,3] ?

Reply:
just to test loading from an `H5T_ARRAY` with rank > 1.

Reply:
I see this number is linked to L60

Reply:
yeah that's right